include/lbann/layers/transform/noise.hpp

```diff
@@ -87,68 +57,23 @@ class noise_layer : public transform_layer {
      return s.str();
   }
 
-  noise_layer* copy() const override { return new noise_layer(*this); }
-
-  std::string get_type() const override { return "noise"; }
-
-  virtual inline void initialize_distributed_matrices() {
-    transform_layer::initialize_distributed_matrices<T_layout>();
-  }
-  data_layout get_data_layout() const override { return T_layout; }
-
-
-  void setup_gpu() override {
-    transform_layer::setup_gpu();
-  #ifndef LBANN_HAS_CUDNN
-    throw lbann_exception("noise_layer: cuDNN not detected");
-  #else
-    m_copy_bp_output_from_gpus = true;
-
-  #endif // #ifndef LBANN_HAS_CUDNN
-  }
-
-  protected:
-  /** noise factor */
-  float m_noise_factor;
+ protected:
 
   void fp_compute() override {
-    if(this->m_using_gpus) {
-      fp_compute_cudnn();
-    } else {
-      fp_compute_cpu();
-    }
+    const auto& input = get_prev_activations();
+    auto& output = get_activations();
+    gaussian_fill(output,
+                  output.Height(), output.Width(),
+                  DataType(0), m_noise_factor);
+    El::Axpy(DataType(1), input, output);
   }
 
   void bp_compute() override {
-    if(this->m_using_gpus) {
-      throw lbann_exception("noise_layer: cuDNN not implemented");
-    } else {
-      El::LockedView(*this->m_error_signal_v, *this->m_prev_error_signal_v);
-    }
-  }
-
-  void fp_compute_cudnn() {
-    throw lbann_exception("noise_layer: cuDNN not implemented");
-  }
-
-  void fp_compute_cpu() {
-    El::Copy(*this->m_prev_activations_v, *this->m_activations_v);
-    AbsDistMat* noise_mat = this->m_activations_v->Construct(this->m_activations_v->Grid(),
-                                                             this->m_activations_v->Root());
-    El::Gaussian(*noise_mat, this->m_activations_v->Height(), 
-                 this->m_activations_v->Width(),
-                 DataType(0), DataType(1));
-    El::Axpy(m_noise_factor,*noise_mat,*this->m_activations_v);
-    
-    //@todo - clip to min and max of input entry
-    auto clip = [&](const DataType& z) { 
```

Comment:
Did we intend to lose the clip function on the entries?

Reply:
This functionality is non-standard and makes the name "noise_layer" misleading. If we want this, we should implement a clip layer.